### PR TITLE
Move batch loading to a separate module

### PR DESCRIPTION
**Pull request by tmcgroul**

### CODE REVIEW

File: src/archive/traits.rs [hunk 1]
```diff
@@ -1,31 +1,12 @@
...
 pub trait ArchiveService {
-    type EvmLogSelection;
-    type EthTransactSelection;
-    type ContractsEventSelection;
-    type GearMessageEnqueuedSelection;
-    type GearUserMessageSentSelection;
-    type EventSelection;
-    type CallSelection;
     type Batch;
+    type BatchOptions;
```
Comment (dmitrylavrenov) [inline src/archive/traits.rs:4]: What do you think about using this name?
```suggestion
    type BatchSelectors;
```

**Review by dmitrylavrenov — APPROVED** [review]
I like it to make one step more to flexible abstractions!

File: src/archive/traits.rs (hunk 1 above)
Comment (tmcgroul) [reply src/archive/traits.rs:4]: BatchOptions has a few non-selector fields like: limit, from_block, to_block
so i'd like to use more general name

Comment (dmitrylavrenov) [reply src/archive/traits.rs:4]: Got it!

**Pull request merged**